[docs] Convert serve/llm/quick-start.rst to MyST Markdown

kouroshHakha opened the pull request:
Convert the Ray Serve LLM quickstart documentation from RST to MyST Markdown format.

Changes:
- Convert quick-start.rst to quick-start.md using MyST syntax
- Apply Ray docs style guide (contractions, active voice, sentence case)
- Convert all RST directives to MyST equivalents (tab-sets, code blocks, cross-refs)
- No content changes, pure format conversion
- Verified: builds successfully with no errors or warnings, formatting has remained the same.

kouroshHakha committed:
[docs] Convert serve/llm/quick-start.rst to MyST Markdown

Convert the Ray Serve LLM quickstart documentation from RST to MyST Markdown format.

Changes:
- Convert quick-start.rst to quick-start.md using MyST syntax
- Apply Ray docs style guide (contractions, active voice, sentence case)
- Convert all RST directives to MyST equivalents (tab-sets, code blocks, cross-refs)
- No content changes, pure format conversion
- Verified: builds successfully with no errors or warnings

This is Phase 0 of the LLM docs reorganization plan.

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

dstrodtman reviewed (approved):
stamped but not built or reviewed

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+
+## Advanced usage patterns
+
+For each usage pattern, we provide a server and client code snippet.
```
```suggestion
For each usage pattern, Ray Serve LLM provides a server and client code snippet.
```

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+You can generate embeddings by selecting the embed task in the engine arguments. Models supporting this use case are listed at [vLLM text embedding models](https://docs.vllm.ai/en/stable/models/supported_models.html#text-embedding-task-embed).
+
+
+Note: You need to set the `VLLM_USE_V1` environment variable to `0`, since the VLLM V1 doesn't yet fully support the embedding endpoints.
```
```suggestion
Note: You need to set the `VLLM_USE_V1` environment variable to `0`, because the VLLM V1 doesn't yet fully support the embedding endpoints.
```

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+# Initialize client
+client = OpenAI(base_url="http://localhost:8000/v1", api_key="fake-key")
+
+# Define a pydantic model of a preset of allowed colors
```
```suggestion
# Define a Pydantic model of a preset of allowed colors
```

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+::::
+
+
+### Serving small models on fraction of GPUs
```
```suggestion
### Serving small models on fractions of GPUs
```

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+
+In this example, we've specified 8 replicas of the model, and we use the `placement_group_config` to specify the fraction of the GPU that each model will use in placement group. We also need to set the `VLLM_RAY_PER_WORKER_GPUS` environment variable so that the vLLM GPU workers will each claim a fraction of the GPU. There's also a resource contention [issue](https://github.com/vllm-project/vllm/i ...
+
+We need to make sure `gpu_memory_utilization` is set to a reasonable value, because vLLM will pre-allocate the GPU memory based on GPU memory utilization, regardless of how we schedule the Ray GPU workers. In this example, setting it to 0.4 means that vLLM will target to use 40% of the GPU memory for the model, its kv-cache, CUDAGraph memory, etc.
```
```suggestion
You need to make sure to set `gpu_memory_utilization` to a reasonable value, because vLLM pre-allocates the GPU memory based on GPU memory utilization, regardless of how you schedule the Ray GPU workers. In this example, setting it to 0.4 means that vLLM targets to use 40% of the GPU memory for the model, its kv-cache, CUDAGraph memory, etc.
```

angelinalg commented on doc/source/serve/llm/quick-start.md:
```diff
@@ -0,0 +1,916 @@
+- Initial replicas count.
+- GPU type used and number of GPUs used.
+
+If you'd like to opt-out from usage data collection, you can follow {ref}`Ray usage stats <ref-usage-stats>` to disable it.
```
```suggestion
To opt-out of usage data collection, you can follow {ref}`Ray usage stats <ref-usage-stats>` to disable it.
```

angelinalg reviewed (approved):
Mostly style nits if you're open to them.

kouroshHakha committed:
Update doc/source/serve/llm/quick-start.md

Co-authored-by: angelinalg <122562471+angelinalg@users.noreply.github.com>
Signed-off-by: kourosh hakhamaneshi <31483498+kouroshHakha@users.noreply.github.com>

kouroshHakha committed:
Address style comments from PR review

- Change 'we provide' to 'Ray Serve LLM provides' for consistency
- Update 'We need to make sure' to 'You need to make sure' with active voice
- Fix 'Huggingface' to 'Hugging Face' (proper capitalization)
- Change 'We collect' to 'The Ray Team collects' for clarity

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
Merge branch 'kh/quickstart-rst-to-md' of https://github.com/kouroshHakha/ray into kh/quickstart-rst-to-md

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
Merge branch 'master' into kh/quickstart-rst-to-md

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
